src/Commands/Plugin/PluginBaseCommand.php

```diff
@@ -0,0 +1,318 @@
+<?php
+
+namespace Pantheon\Terminus\Commands\Plugin;
+
+use Pantheon\Terminus\Commands\TerminusCommand;
+use Pantheon\Terminus\Exceptions\TerminusNotFoundException;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Finder\Finder;
+
+/**
+ * Class PluginBaseCommand
+ * Base class for Terminus commands that deal with sending Plugin commands
+ * @package Pantheon\Terminus\Commands\Plugin
+ */
+abstract class PluginBaseCommand extends TerminusCommand
+{
+    /**
+     * PluginBaseCommand constructor.
+     */
+    public function __construct()
+    {
+         // Check for minimum plugin command requirements.
+         $windows = (php_uname('s') == 'Windows NT');
+         $test_command = $windows ? 'where' : 'command -v';
+         $file = popen("$test_command git", 'r');
+         $result = fgets($file, 255);
+         $valid = $windows ? !preg_match('#Could not find files#', $result) : !empty($result);
+         if (!$valid) {
+            die("Please install the git command to enable plugin management.\n");
```

Comment:
Never use `die()` -- this method is not testable. Instead, typically you should throw an exception (e.g. TerminusException). However, you should not fail like this in a command constructor, as failures here will prevent the operation of any command.